pkg/downloader/manager.go

```diff
@@ -64,42 +66,102 @@ type Manager struct {
 // If the lockfile is not present, this will run a Manager.Update()
 //
 // If SkipUpdate is set, this will not update the repository.
-func (m *Manager) Build() error {
+func (m *Manager) Build() ([]*chartutil.Dependency, error) {
 	c, err := m.loadChartDir()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// If a lock file is found, run a build from that. Otherwise, just do
 	// an update.
 	lock, err := chartutil.LoadRequirementsLock(c)
 	if err != nil {
-		return m.Update()
+		err = m.Update()
+		if err != nil {
+			return nil, err
+		}
+		req, err := chartutil.LoadRequirements(c)
+		if err != nil {
+			if err == chartutil.ErrRequirementsNotFound {
+				return nil, nil
+			}
+			return nil, err
+		}
+		return req.Dependencies, nil
 	}
 
 	// A lock must accompany a requirements.yaml file.
 	req, err := chartutil.LoadRequirements(c)
 	if err != nil {
-		return fmt.Errorf("requirements.yaml cannot be opened: %s", err)
+		return nil, fmt.Errorf("requirements.yaml cannot be opened: %s", err)
 	}
 	if sum, err := resolver.HashReq(req); err != nil || sum != lock.Digest {
-		return fmt.Errorf("requirements.lock is out of sync with requirements.yaml")
+		return nil, fmt.Errorf("requirements.lock is out of sync with requirements.yaml")
 	}
 
 	// Check that all of the repos we're dependent on actually exist.
 	if err := m.hasAllRepos(lock.Dependencies); err != nil {
-		return err
+		return nil, err
 	}
 
 	if !m.SkipUpdate {
 		// For each repo in the file, update the cached copy of that repo
 		if err := m.UpdateRepositories(); err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	// Now we need to fetch every package here into charts/
-	return m.downloadAll(lock.Dependencies)
+	return req.Dependencies, m.downloadAll(lock.Dependencies)
+}
+
+// BuildRecursively rebuilds recursively a local charts directory from a lockfile.
+// If the lockfile is not present, this will run a Manager.Update()
+//
+// If SkipUpdate is set, this will not update the repository.
+func (m *Manager) BuildRecursively() error {
+	// Build the main chart
+	dependencies, err := m.Build()
+	if err != nil {
+		return err
+	}
+
+	m.SkipUpdate = true
```

Comment:
Nit: I would recommend adding a comment here for anyone else who edits this code in the future about why `SkipUpdate` is set to true

Reply:
I would also set `SkipUpdate` back to its original value at the end of this function. I am afraid of weird side effects down the road